controllers/operatorpolicy_controller.go

```diff
@@ -108,167 +101,300 @@ func (r *OperatorPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
+) (*operatorv1.OperatorGroup, error) {
+	operatorGroup := new(operatorv1.OperatorGroup)
 
-				return reconcile.Result{}, err
-			}
+	operatorGroup.Status.LastUpdated = &metav1.Time{} // without this, some conversions can panic
+	operatorGroup.SetGroupVersionKind(operatorGroupGVK)
 
-			// Created successfully, requeue result
-			r.setCompliance(ctx, policy, policyv1.NonCompliant)
+	sub := make(map[string]interface{})
 
-			return reconcile.Result{Requeue: true}, err
-		} else if policy.Spec.OperatorGroup != nil {
-			// If inform mode, keep going and return at the end without requeue. Before
-			// continuing, should check if required og spec is created to set compliance state
+	err := json.Unmarshal(policy.Spec.Subscription.Raw, &sub)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshalling subscription: %w", err)
+	}
 
-			// Set to non compliant because operatorgroup does not exist on cluster, but
-			// should still go to check subscription
-			r.setCompliance(ctx, policy, policyv1.NonCompliant)
-		}
+	subNamespace, ok := sub["namespace"].(string)
+	if !ok {
+		return nil, fmt.Errorf("namespace is required in spec.subscription")
 	}
 
-	// Create new subscription
-	if cachedSubscription == nil {
-		if strings.EqualFold(string(policy.Spec.RemediationAction), string(policyv1.Enforce)) {
-			subscriptionSpec := buildSubscription(policy)
+	// Create a default OperatorGroup if one wasn't specified in the policy
+	if policy.Spec.OperatorGroup == nil {
+		operatorGroup.ObjectMeta.SetNamespace(subNamespace)
+		operatorGroup.ObjectMeta.SetGenerateName(subNamespace + "-") // This matches what the console creates
+		operatorGroup.Spec.TargetNamespaces = []string{}
 
-			err := r.Create(ctx, subscriptionSpec)
-			if err != nil {
-				OpLog.Error(err, "Could not handle missing musthave object")
-				r.setCompliance(ctx, policy, policyv1.NonCompliant)
+		return operatorGroup, nil
+	}
 
-				return reconcile.Result{}, err
-			}
+	opGroup := make(map[string]interface{})
 
-			// Future work: Check availability/status of all resources
-			// managed by the OperatorPolicy before setting compliance state
-			r.setCompliance(ctx, policy, policyv1.Compliant)
+	err = json.Unmarshal(policy.Spec.OperatorGroup.Raw, &opGroup)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshalling operatorGroup: %w", err)
+	}
 
-			return reconcile.Result{}, nil
-		}
-		// inform
-		r.setCompliance(ctx, policy, policyv1.NonCompliant)
+	// Fallback to the Subscription namespace if the OperatorGroup namespace is not specified in the policy.
+	ogNamespace := subNamespace
+
+	if specifiedNS, ok := opGroup["namespace"].(string); ok {
+		ogNamespace = specifiedNS
 	}
 
-	// Will only reach this if in inform mode
-	return reconcile.Result{}, nil
+	name, ok := opGroup["name"].(string)
+	if !ok {
+		return nil, fmt.Errorf("name is required in operatorGroup.spec")
+	}
+
+	spec := new(operatorv1.OperatorGroupSpec)
+
+	err = json.Unmarshal(policy.Spec.OperatorGroup.Raw, spec)
```

Comment:
If you follow my other comments, you can use unstructured for the conversion but it does create an extra json.Marshal that your implementation does not.
```
err = k8sruntime.DefaultUnstructuredConverter.FromUnstructured(opgroup.UnstructuredContent(), &spec)
```